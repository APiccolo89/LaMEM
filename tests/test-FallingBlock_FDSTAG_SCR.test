EXEC            = ../bin/LaMEM -ParamFile ./input/FallingBlock_SCR.dat -vpt_element FDSTAG -AddRandomNoiseParticles 0 -time_end 1 -BC.UpperBound 1
NPROCS          = 1
TEST_TYPE       = tolerance_diff_pl,1.0e-4
