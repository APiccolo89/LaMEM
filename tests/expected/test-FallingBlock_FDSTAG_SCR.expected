# -------------------------------------------------------------------------- 
#                   Lithosphere and Mantle Evolution Model                   
#              Current Revision: 4812 - 2013-10-01 23:57:18		     
#  Modified items: LaMEM_Particles.c, Elements.c, Assembly_FDSTAG.c, fdstag/JacRes.c, 
#     Compiled: Date: Oct  9 2013 - Time: 23:02:54 - Mode:  	    
# -------------------------------------------------------------------------- 
 Parsing input file : ./input/FallingBlock_SCR.dat 
# Adding PetscOption: -vpt_element Q1P0 
# Adding PetscOption: -restart 0 
# Adding PetscOption: -save_breakpoints 0 
# Adding PetscOption: -scr_ksp_rtol 1.0e-4 
# Adding PetscOption: -scr_ksp_atol 1.0e-13 
# Adding PetscOption: -scr_pc_type bjacobi 
# Adding PetscOption: -vs_ksp_type fgmres 
# Adding PetscOption: -vs_pc_type fieldsplit 
# Adding PetscOption: -vs_fieldsplit_ksp_type fgmres 
# Adding PetscOption: -vs_fieldsplit_ksp_rtol 1.0e-2 
# Adding PetscOption: -vs_pc_fieldsplit_block_size 3 
# Adding PetscOption: -vs_pc_fieldsplit_type SYMMETRIC_MULTIPLICATIVE 
# LithColDepth[0] = 0 
# LithColDens[0] = 0 
# LithColDens[1] = 0 
# --------------------------------------------------------------------------
# Phase material parameters read from ./input/FallingBlock_SCR.dat: 
#    phase_1 (id=1), VISCOSITY, constant [0,0,0] = 1.000000e+03 
#    phase_1 (id=1), ELASTICITY, constant [0,1,0] = 5.000000e+10, 1.000000e+00 
#    phase_1 (id=1), DENSITY, temperature-dependent [0,2,0] = 2.000000e+00, 0.000000e+00, 0.000000e+00 
#    phase_1 (id=1), ENERGY, constant [0,3,0] = 3.000000e+00 1.050000e+03 0.000000e+00 0.000000e+00 
#    phase_0 (id=0), VISCOSITY, constant [1,0,0] = 1.000000e+00 
#    phase_0 (id=0), ELASTICITY, constant [1,1,0] = 5.000000e+10, 1.000000e+00 
#    phase_0 (id=0), DENSITY, temperature-dependent [1,2,0] = 1.000000e+00, 0.000000e+00, 0.000000e+00 
#    phase_0 (id=0), ENERGY, constant [1,3,0] = 3.000000e+00 1.050000e+03 0.000000e+00 0.000000e+00 
 Stokes solver             : Schur Complement Reduction
 Velocity solver           : User-defined (set options with prefix vs_)
 Note: Galerkin geometric multigrid is deactivated for user-defined solver 
 Element-type used         : FDSTAG;  [Q1P0 Q2Pm1_global Q2Pm1_local Q1Q1 FDSTAG] (change with -vpt_element)
 Total # of cpu's          : 1 
 Total # of velocity dof's : 2187 
 Computing mean material per element using arithmetic averaging from integration points 
 Employing code in an Eulerian manner 
 Gravity angle with z-axis =  0 
 Approximate Aspect ratio of cells [dx/dz, dy/dz] = [1 1] 
 Number of tracers/cell in [x,y,z]-direction = [2,2,2] 
 NumSurfaceNodes=1  nz=9 
 # Not employing nonlinear iterations 
 BC employed: BC.[Left Right; Front Back; Lower Upper]=[1 1; 1 1; 1 1] 
# Initial time step: 0.001 years 
 Input is assumed to be in non-dimensional units 
 Fine Grid parameters, nel=[8,8,8], cpu=[1,1,1] 
# Finished initializing Stokes matrices : 0.0038352 s 
# Not adding random noise to initial particle distribution. 
Initialized particles 
#rank=0, Attributing 4096 particles to elements took 0.00264597 s 
The total # of particles present is 4096 
 *** Using a falling block setup *** 
 Falling block coordinates:  [left,right]=[0.250000,0.750000]; [front,back]=[0.250000,0.750000]; [bottom,top]=[0.250000,0.750000] 
Saved initial particle distribution to directory ./InitialParticles 
The total memory used by integration points is 1.351680e-01 MB
The max memory used by integration points is 1.351680e-01 MB
The min memory used by integration points is 1.351680e-01 MB
#  Computing material properties from particles for FDSTAG ...   [0.0033052 s] 
#  Interpolating meshes and temperatures took 0.00351381 s 
#  Forming FD stiffness matrix mx,my,mz=(9, 9, 9) ...  
#  FDSTAG: Time to insert stencil into operators 1.9901e-03 
#  FDSTAG: Time to insert bc's into operators 6.9141e-05 
#  FDSTAG: Time to assemble operators 1.4210e-04 
#  Forming stiffness matrixes took 0.00270486 s 
SCR: Viscosity structure info: min(0)=1.000000e+00 : max(146)=1.000000e+03 : contrast=1.000000e+03 
  Residual norms for scr_ solve.
  0 KSP Residual norm 3.633873197835e+00 
  1 KSP Residual norm 2.143791421152e-01 
  2 KSP Residual norm 9.977482011634e-02 
  3 KSP Residual norm 4.381736613281e-02 
  4 KSP Residual norm 1.337228628820e-02 
  5 KSP Residual norm 1.094377898902e-02 
  6 KSP Residual norm 3.469734726849e-03 
  7 KSP Residual norm 2.113818600494e-03 
  8 KSP Residual norm 8.533467987592e-04 
  9 KSP Residual norm 7.252700720707e-04 
 10 KSP Residual norm 1.854109023321e-04 
Linear solve converged due to CONVERGED_RTOL iterations 10
# Time: p solve = 9.53e-02 
# Time: u solve = 1.39e-02 
#  Finished Stokes solve after 0.129203 s 
------------------------------------------
Stokes solver summary: 
  divergence: 
    Div_min = 1.139781708598e-08 
    Div_max = 3.784069644194e-05 
    |Div|_2 = 2.221385913795e-04 
    |Div|_1 = 3.781590371688e-03 
  momentum: 
    mRes_min = 0.000000000000e+00 
    mRes_max = 2.952274864132e-05 
    |mRes|_2 = 1.457103587679e-04 
    |mRes|_1 = 2.421419627402e-03 
  velocity: 
    u_min   = -2.092742534529e-03 
    u_max   = 3.787578381498e-03 
    |u|_2   = 4.127909329866e-02 
    |u|_1   = 1.080007271146e+00 
  pressure: 
    p_min   = -5.271158694972e-01 
    p_max   = 5.271162421987e-01 
    |p|_2   = 7.526375654419e+00 
    |p|_1   = 1.477381456019e+02 
------------------------------------------
The total memory used by integration points is 1.351680e-01 MB
The max memory used by integration points is 1.351680e-01 MB
The min memory used by integration points is 1.351680e-01 MB
#  Solving for velocity took 0.132972 s 
#  Forming FD stiffness matrix for energy equation with mx,my,mz=(9, 9, 9) with Top/Bottom T=[0,1]...  
#  Forming temperature stiffness matrixes took 0.000235081 s 
#  Solving for temperature took 0.000433922 s 
The total memory used by integration points is 1.351680e-01 MB
The max memory used by integration points is 1.351680e-01 MB
The min memory used by integration points is 1.351680e-01 MB
The total memory used by particles is 1.146880e+00 MB
The max memory used by particles is 1.146880e+00 MB
The min memory used by particles is 1.146880e+00 MB
#  Updating properties @ integration points and particles took 0.00693989 s 
AdvectionMethod = 0 
  Starting particle advection 
  Finished particle advection 
Minimum z-level phase 1 0.281248 
# Writing output to existing directory Timestep_000000 
# Saved MATLAB output in file Timestep_000000/FB_test.0.1000000.out 
# Saved Paraview VTS Velocity output in file FB_test_velocity_step000000-mesh.pvts 
# Saved Paraview VTS Temperature output in file FB_test_temperature_step000000-mesh.pvts 
# Saved Paraview VTS Surface Topography output in file FB_test_Surface_topography_step000000-mesh.pvts 
# Saved Paraview VTS Bottom Topography output in file FB_test_Bottom_topography_step000000-mesh.pvts 
# Saved Paraview VTS point-wise quadrature output in file FB_test_quadrature_step000000-mesh.pvts 
 Starting GetParticleNodes 
#rank=0, Attributing 4096 particles to elements took 0.00195384 s 
The total # of particles present is 4096 
 Finished GetParticleNodes 
 Maximum velocity  0.00378758    
 Time = 0.001, dt=1e-05   
# Finished timestep 0 out of 1 in 0.15252 s
  
# Total time required: 0.169768 s 
#PETSc Option Table entries:
-AddRandomNoiseParticles 0
-BC.UpperBound 1
-ParamFile ./input/FallingBlock_SCR.dat
-options_left
-restart 0
-save_breakpoints 0
-scr_ksp_atol 1.0e-13
-scr_ksp_converged_reason
-scr_ksp_max_it 1000
-scr_ksp_monitor
-scr_ksp_rtol 1.0e-4
-scr_ksp_type fgmres
-scr_pc_type bjacobi
-time_end 1
-vpt_element FDSTAG
-vs_fieldsplit_ksp_rtol 1.0e-2
-vs_fieldsplit_ksp_type fgmres
-vs_ksp_type fgmres
-vs_pc_fieldsplit_block_size 3
-vs_pc_fieldsplit_type SYMMETRIC_MULTIPLICATIVE
-vs_pc_type fieldsplit
#End of PETSc Option Table entries
There are no unused options.
